### PR TITLE
Support DeviceCgroupRules to actually get added.

### PR DESCRIPTION
**Pull request by rhatdan**
Fixes: https://github.com/containers/podman/issues/10302

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by TomSweeneyRedHat** [comment]
Lots of Red test unhappiness.
The code LGTM

File: pkg/specgen/specgen.go [hunk 1]
```diff
@@ -239,6 +239,9 @@ type ContainerStorageConfig struct {
 	// Devices are devices that will be added to the container.
 	// Optional.
 	Devices []spec.LinuxDevice `json:"devices,omitempty"`
+	// DeviceCGroupRule are device cgroup rules that allow containers
+	// to use additional types of devices.
+	DeviceCGroupRule []string `json:"device_cgroup_rule,omitempty"`
```
Comment (mheon) [inline pkg/specgen/specgen.go:244]: Can we do the parsing on the frontend, so we can pass a struct here (including the type, path, and access permissions)?

Comment (rhatdan) [reply pkg/specgen/specgen.go:244]: That would require all users of the API to do the processing. I believe this is something that Docker supports as well.
WDYT @jwhonce @baude

Comment (mheon) [reply pkg/specgen/specgen.go:244]: Generally speaking, we try to parse in the frontend to the greatest extent possible, so the server does not have to do much validation of the contents of specgen.

**Comment by TomSweeneyRedHat** [comment]
LGTM
But I'd like @mheon to eyeball and merge if his concern has been answered.

**Comment by cdoern** [comment]
LGTM, just a general note: wondering if in the future we should look to combine all `[]spec.` elements into a singular entity since one spec entity should contain all of the things we are referencing?

**Pull request merged**